### DISCUSSION TITLE
different between these two instance of FundMe

### DISCUSSION THREAD

**Discussion by king-null** [Cyfrin/foundry-full-course-cu#1526] [Q&A]
i understand that both is declared but only one is initialized, but i dont understand the work and the limit if you dont initialize it 

```solidity
FundMe public fundMe;
```
AND

```solidity
FundMe fundMe = new FundMe();
```
also i want to add that i'm finding it confusing understanding where the function activeNetworkConfig(); , i just want to clarify if is a function from the Script contract
```
these below is the code associated with the activeNetworkConfig(); 

```solidity
contract DeployFundMe is Script {
    function run() external returns (FundMe) {    
        HelperConfig helperConfig = new HelperConfig();
        address ethUsdPriceFeed = helperConfig.activeNetworkConfig();
```

**Comment by EngrPips** [comment]
Hello @king-null , we always do `FundMe public fundMe;` outside of any function and in the main body of the contract if we want to have the contract object available in storage which is the global state of the contract, which simply means you can now access the `fundMe` object from any function inside the contract and we would always do `FundMe fundMe = new FundMe();` inside of a particular function of a contract so as to create a quick contract object of the type `FundMe` which we can interact with.

About `activeNetworkConfig()` thats a public state variable which is available on your `HelperConfig.s.sol` , originally we declared a variable name `activeNetworkConfig` like this `NetworkConfig public activeNetworkConfig` but we are able to use the `activeNetworkConfig()` function to get that variable because soildity automatically creates a getter function for state variables in a contract. 

I hope explained well enough to make you understand, if anything still seem confusing you should please ask

**Comment by Niferu** [comment]
Hello @king-null 

Most simple explanation is that

```solidity
// `FundMe` here is just a type, so we are creating unassigned variable named `fundMe` of type `FundMe` (which is contract type)
FundMe public fundMe;

// This line gives us instance of our contract saved under `fundMe` variable of type `FundMe`. This instance is different as we are already assigning new contract to it
FundMe fundMe = new FundMe(); 
```
